internal/radix/adaptive/node.go

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package adaptive
+
+import "sync"
+
+type Node[T any] interface {
+	getPartialLen() uint32
+	setPartialLen(uint32)
+	getArtNodeType() uint8
+	setArtNodeType(uint8)
+	getNumChildren() uint8
+	setNumChildren(uint8)
+	getPartial() []byte
+	setPartial([]byte)
+	isLeaf() bool
+	Iterator() *Iterator[T]
+	PathIterator([]byte) *PathIterator[T]
+	matchPrefix([]byte) bool
+	getChild(int) *Node[T]
+	setMutex(*sync.RWMutex)
+	getMutex() *sync.RWMutex
+	Clone() *Node[T]
+}
```

Comment:
```suggestion
type Node[T any] interface {
	getPartialLen() uint32
	setPartialLen(uint32)
	getArtNodeType() nodeType
	setArtNodeType(nodeType)
	getNumChildren() uint8
	setNumChildren(uint8)
	getPartial() []byte
	setPartial([]byte)
	isLeaf() bool
	Iterator() *Iterator[T]
	PathIterator([]byte) *PathIterator[T]
	matchPrefix([]byte) bool
	getChild(int) *Node[T]
	Clone() *Node[T]
}
```

There are a couple changes in here:

1. Change from using a `uint8` to `nodeType` (this might be a uint8 under the hood but its strongly typed so the compiler can help us ensure we don't use invalid values accidentally).
2. Get rid of the mutexes as these really shouldn't be needed if the tree is immutable.